lib/matplotlib/pyplot.py

```diff
@@ -2084,7 +2084,7 @@ def set_cmap(cmap):
     matplotlib.cm.register_cmap
     matplotlib.cm.get_cmap
     """
-    cmap = cm.get_cmap(cmap)
+    cmap = colormaps[cmap]
```

Comment:
Should this not use `_ensure_cmap` to accept a `Colormap` instance?

Reply:
It should.

Reply:
sigh, there are exception stability issues, this used to raise `ValueError` but `_ensure_cmap` will raise `KeyError`.